pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -195,32 +195,33 @@ def __init__(self, *args, **kwargs):
 
         self._nstages = len(self.b)
 
-        # Register a kernel to multiply rhs with local pseudo time-step
+        # Allocate storage for the local pseudo time-step field
+        self.dtau_upts = [self.backend.matrix(shape, np.ones(shape)*self._dtau,
+                                              tags={'align'})
+                          for shape in self.system.ele_shapes]
+
+        # Register a pointwise kernel for the low-storage stepper
         self.backend.pointwise.register(
-            'pyfr.integrators.dual.pseudo.kernels.localdtau'
+            'pyfr.integrators.dual.pseudo.kernels.rkvdh2pseudo'
         )
 
-        tplargs = dict(ndims=self.system.ndims, nvars=self.system.nvars)
-
-        self.dtau_upts = proxylist([])
-        for ele, shape in zip(self.system.ele_map.values(),
-                              self.system.ele_shapes):
-            # Allocate storage for the local pseudo time-step
-            dtaumat = self.backend.matrix(shape, np.ones(shape)*self._dtau,
-                                          tags={'align'})
-            self.dtau_upts.append(dtaumat)
-
-            # Append the local dtau kernels to the proxylist
-            self.pintgkernels['localdtau'].append(
-                self.backend.kernel(
-                    'localdtau', tplargs=tplargs, dims=[ele.nupts, ele.neles],
-                    negdivconf=ele.scal_upts_inb, dtau_upts=dtaumat
-                )
+    @memoize
+    def _get_rkvdh2pseudo_kerns(self, stage, r1, r2, rold, rerr=None):
+        kerns = []
+        tplargs = {'a': self.a, 'b': self.b, 'e': self.e, 'stage': stage,
+                   'nstages': self._nstages, 'nvars': self.system.nvars,
+                   'errest': rerr is not None}
+
+        for dims, em, dtaum in zip(self.system.ele_shapes,
```

Comment:
I try to avoid line breaks in for loops where possible.  Lets see if there is a change we can make here.

Reply:
I can do
```
zipped = zip(self.system.ele_shapes, ....)
for dims, em, dtau in zipped:
```
What do you think?

Reply:
Probably want a better name than zipped, but I think the idea is sound.

Reply:
Okay